src/lazy/text/matched.rs

```diff
@@ -451,3 +455,290 @@ impl MatchedSymbol {
+
+        if self.precision == TimestampPrecision::HourAndMinute {
+            if let Some(offset) = offset_minutes {
+                return timestamp.with_offset(offset).build();
+            } else {
+                return timestamp.build();
+            }
+        }
+
+        let second_text = matched_input.slice(17, 2).as_text().unwrap();
+        let seconds = u32::from_str(second_text).unwrap();
+        let timestamp = timestamp.with_second(seconds);
+
+        // There's guaranteed to be a 19th byte. It will either be a `.`, indicating the beginning
+        // of fractional seconds or a `+`/`-`/`Z` comprising part of the offset.
+        if matched_input.bytes()[19] != b'.' {
+            // There are no fractional seconds; we can apply the offset (if any) and return.
+            if let Some(offset) = offset_minutes {
+                return timestamp.with_offset(offset).build();
+            } else {
+                return timestamp.build();
+            }
+        }
+
+        // There are fractional seconds. We need to scan ahead to the first non-digit byte in the
+        // input following the leading `.`.
+        let fractional_start = 20;
+        let mut fractional_end = fractional_start;
+        for byte in matched_input.slice_to_end(fractional_start).bytes() {
+            if !byte.is_dec_digit() {
+                break;
+            }
+            fractional_end += 1;
+        }
+        let fractional_text = matched_input
+            .slice(fractional_start, fractional_end - fractional_start)
+            .as_text()
+            .unwrap();
+        let timestamp = match fractional_text.len() {
+            3 => {
+                let milliseconds = u32::from_str(fractional_text).unwrap();
+                timestamp.with_milliseconds(milliseconds)
+            }
+            6 => {
+                let microseconds = u32::from_str(fractional_text).unwrap();
+                timestamp.with_microseconds(microseconds)
+            }
+            9 => {
+                let nanoseconds = u32::from_str(fractional_text).unwrap();
+                timestamp.with_nanoseconds(nanoseconds)
+            }
+            _ => {
+                // For less common precisions, store a Decimal
+                let coefficient = BigUint::from_str(fractional_text).unwrap();
+                let decimal = Decimal::new(coefficient, -(fractional_text.len() as i64));
+                timestamp.with_fractional_seconds(decimal)
+            }
+        };
+
+        if let Some(offset) = offset_minutes {
+            timestamp.with_offset(offset).build()
+        } else {
+            timestamp.build()
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum MatchedTimestampOffset {
+    Zulu,
+    HoursAndMinutes(MatchedHoursAndMinutes),
+    Unknown,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct MatchedHoursAndMinutes {
+    is_negative: bool,
+    hours_offset: usize,
```

Comment:
Do you mean minutes?
```suggestion
    minutes_offset: usize,
```

Reply:
`hours` is correct; this is the text offset of the first `H` in the offset substring `HH:MM`. I'll add a comment to clarify!